CMakeLists.txt

```diff
@@ -58,6 +58,24 @@ __sync_add_and_fetch(&a, 1);
 return 0;
 }" ARCH_SUPPORTS_64BIT_ATOMICS)
 
+FIND_LIBRARY(LIBC_BACKTRACE_LIB "execinfo")
+IF (LIBC_BACKTRACE_LIB)
+    SET(CMAKE_REQUIRED_LIBRARIES ${LIBC_BACKTRACE_LIB})
+    LIST(APPEND EXTRA_LIBS ${LIBC_BACKTRACE_LIB})
+ENDIF()
+CHECK_C_SOURCE_COMPILES("
+#include <execinfo.h>
+int main(void) {
+void *p[10];
+int ret = backtrace(p, 10);
+backtrace_symbols_fd(p, ret, 2);
+return 0;
+}" LIBC_HAS_BACKTRACE)
+
+IF (LIBC_HAS_BACKTRACE)
+    ADD_DEFINITIONS("-DLIBC_HAS_BACKTRACE")
+ENDIF ()
```

Comment:
Would it be possible to surround all the logic that deals with backtrace with `CMakePushCheckState` / `CMakePopCheckState` so that we would not have side-effects?

Reply:
I didn't know about those functions, that's handy, thanks for pointing this out! Addressed in 52d8060